src/proj_strtod.c

```diff
@@ -0,0 +1,251 @@
+        p++;
+
+    /* Empty string? */
+    if (0==*p) {
+        errno = EINVAL;
+        if (endptr)
+            *endptr = p;
+        return HUGE_VAL;
+    }
+
+    /* Then handle optional prefixed sign */
+    switch (*p) {
+        case '-':
+            sign = -1,  p++;  break;
+        case '+':
+            sign =  1,  p++;  break;
+        default:
+            if (isdigit(*p) || '_'==*p || '.'==*p)
+                break;
+            if (endptr)
+                *endptr = p;
+            errno = EINVAL;
+            return HUGE_VAL;
+    }
+
+    /* Now expect a (potentially zero-length) string of digits */
+    while (isdigit(*p) || ('_'==*p)) {
+        if ('_'==*p) {
+            p++;
+            continue;
+        }
+        number = number * 10. + (*p - '0');
+        p++;
+        num_digits_total++;
+    }
+        
+    /* Do we have a fractional part? */
+    if ('.'==*p) {
+        p++;
+
+        while (isdigit(*p) || '_'==*p) {
+            if ('_'==*p) {
+                p++;
+                continue;
+            }
+            number = number * 10. + (*p - '0');
+            p++;
+            num_digits_total++;
+            num_digits_after_comma++;
+        }
+
+        exponent = -num_digits_after_comma;
+    }
+
+    /* non-digit */
+    if (0==num_digits_total) {
+        errno = EINVAL;
+        if (endptr)
+            *endptr = p;
+        return HUGE_VAL;
+    }
+
+    if (sign==-1)
+        number = -number;
+
+    /* Do we have an exponent part? */
+    if (*p == 'e' || *p == 'E') {
+        p++;
+        /* Does it have a sign? */
+        sign = 0;
+        if ('-'==*p)
+            sign = -1;
+        if ('+'==*p)
+            sign = +1;
+        if (0==sign) {
+            if (!isdigit(*p) && *p!='_') {
+                if (endptr)
+                    *endptr = p;
+                return HUGE_VAL;
+            }
+        }
+        else
+            p++;
+
+
+        /* Go on and read the exponent */
+        n = 0;
+        while (isdigit(*p) || '_'==*p) {
+            if ('_'==*p) {
+                p++;
+                continue;
+            }
+            n = n * 10 + (*p - '0');
+            p++;
+        }
+
+        if (-1==sign)
+            n = -n;
+        exponent += n;
+    }
+
+    if ((exponent < DBL_MIN_EXP) || (exponent > DBL_MAX_EXP)) {
+      errno = ERANGE;
+      if (endptr)
+          *endptr = p;
+      return HUGE_VAL;
+    }
+
+    number *= pow (10, exponent);
+
+    if (fabs(number) > DBL_MAX)
+        errno = ERANGE;
+
+    if (endptr)
+        *endptr = p;
+
+    return number;
+}
+
+double proj_atof(const char *str) {
+  return proj_strtod(str, (void *) 0);
+}
+
+#ifdef TEST
+#include <string.h>
+
+int main (int argc, char **argv) {
+    double res;
+    char *endptr;
+    if (argc < 2)
+        return 0;
+    res =  proj_strtod (argv[1], &endptr);
+    printf ("res = %20.15g. Rest = [%s],  errno = %d\n", res, endptr, (int) errno);
+    return 0;
+}
+#endif
```

Comment:
There's a wierd file ending here.

Reply:
Added cr at end

Reply:
Will remove *.rst files from this PR, and initiate a new one once you push your doc PR.

Reply:
Actually removed apps/cct.rst, and manually removed added material in apps/index.rst - hence, it probably initiates a new doc build